src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -1453,6 +1448,349 @@ static void G()
+                //         (z ?? w).F.ToString();
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "z").WithLocation(16, 10),
+                // (16,15): warning CS8619: Nullability of reference types in value of type 'B<object>' doesn't match target type 'A<object?>'.
+                //         (z ?? w).F.ToString();
+                //Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "w").WithArguments("B<object>", "A<object?>").WithLocation(16, 15),
+                // (16,9): warning CS8602: Possible dereference of a null reference.
+                //         (z ?? w).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "(z ?? w).F").WithLocation(16, 9),
+                // (17,10): warning CS8619: Nullability of reference types in value of type 'B<object>' doesn't match target type 'A<object?>'.
+                //         (w ?? z).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "w").WithArguments("B<object>", "A<object?>").WithLocation(17, 10),
+                // (17,9): warning CS8602: Possible dereference of a null reference.
+                //         (w ?? z).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "(w ?? z).F").WithLocation(17, 9));
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): Conversions: Conversion
+        [Fact]
+        public void ImplicitConversion_NullCoalescingOperator_02()
+        {
+            var source =
+@"interface IIn<in T>
+{
+    void F(T x, T y);
+}
+class C
+{
+    static void F(IIn<object>? x, IIn<string?> y)
+    {
+        (x ?? y).F(string.Empty, null);
+        (y ?? x).F(string.Empty, null);
+    }
+    static void G(IIn<object?> z, IIn<string>? w)
+    {
+        (z ?? w).F(string.Empty, null);
+        (w ?? z).F(string.Empty, null);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (9,10): warning CS8619: Nullability of reference types in value of type 'IIn<object>' doesn't match target type 'IIn<string?>'.
+                //         (x ?? y).F(string.Empty, null);
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IIn<object>", "IIn<string?>").WithLocation(9, 10),
+                // (10,10): hidden CS8607: Expression is probably never null.
+                //         (y ?? x).F(string.Empty, null);
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "y").WithLocation(10, 10),
+                // (10,15): warning CS8619: Nullability of reference types in value of type 'IIn<object>' doesn't match target type 'IIn<string?>'.
+                //         (y ?? x).F(string.Empty, null);
+                //Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IIn<object>", "IIn<string?>").WithLocation(10, 15),
```

Comment:
>//Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("IIn<object>", "IIn<string?>").WithLocation(10, 15), [](start = 16, length = 131)

Why is this warning not reported?

Reply:
`node.Right` is a `BoundConversion` converting `IIn<object>` to `IIn<string?>` and `VisitConversion` does not warn on nested nullability mismatches currently.

---
In reply to: [175238675](https://github.com/dotnet/roslyn/pull/25358#discussion_r175238675) [](ancestors = 175238675)